src/codegen/bound.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{
+    analysis::{
+        bounds::{Bound, BoundType},
+        ref_mode::RefMode,
+    },
+    library::Nullable,
+};
+
+impl Bound {
+    pub(crate) fn trait_bound(&self) -> Option<String> {
+        match self.bound_type {
+            BoundType::NoWrapper => Some(format!("{}: {}", self.alias.unwrap(), self.type_str)),
+            BoundType::AsRef(Some(_ /*lifetime*/)) => panic!("AsRef cannot have a lifetime"),
+            _ => None,
+        }
+    }
+
+    // TODO: Name
+    /// `nullable` only applies to [`BoundType::IsA`]
+    pub(crate) fn use_as_type(
+        &self,
+        ref_mode: RefMode,
+        nullable: Nullable,
+        r#async: bool,
+    ) -> String {
+        // TODO: Should we assert that Some alias implies NoWrapper, and the other way around?
+        // Alternatively, should the alias be encoded in the bound type, as a properly separated
+        // "mode" entirely where type names are _reused_ in the function to call other functions?
```

Comment:
Same question about the heuristic. I'm thinking to split this up to always return the alias when it exists, otherwise return `impl TheTraitBound`.